docs/ocis/development/profiling.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "Profiling"
+date: 2021-08-24T12:32:20+01:00
+weight: 56
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/development
+geekdocFilePath: profiling.md
+---
+
+{{< toc >}}
+
+The only way to enable the profiler currently is to explicitly select which areas to collect samples for. In order to do this, the following steps have to be followed.
+
+## 1. Clone Reva
+
+`git clone github.com/cs3org/reva`
+
+## 2. Patch reva with the area that you want sampled.
+
+For the purposes of this docs let's use the WebDAV `PROPFIND` path.
+
+```diff
+diff --git a/internal/http/services/owncloud/ocdav/propfind.go b/internal/http/services/owncloud/ocdav/propfind.go
+index 0e9c99be..f271572f 100644
+--- a/internal/http/services/owncloud/ocdav/propfind.go
++++ b/internal/http/services/owncloud/ocdav/propfind.go
+@@ -32,6 +32,8 @@ import (
+        "strings"
+        "time"
+
++       _ "net/http/pprof"
++
+        userv1beta1 "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+        rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+        link "github.com/cs3org/go-cs3apis/cs3/sharing/link/v1beta1"
+@@ -311,6 +313,12 @@ func requiresExplicitFetching(n *xml.Name) bool {
+        return true
+ }
+
++func init() {
++       go func() {
++               http.ListenAndServe(":1234", nil)
++       }()
++}
++
+ // from https://github.com/golang/net/blob/e514e69ffb8bc3c76a71ae40de0118d794855992/webdav/xml.go#L178-L205
+ func readPropfind(r io.Reader) (pf propfindXML, status int, err error) {
+        c := countingReader{r: r}
+```
+
+The previous patch will:
+
+1. import `net/http/pprof`, which will register debug handlers in `DefaultServeMux`.
+2. define a `init()` function that starts an http server with the previously registered handlers.
+
+With everything running one should have access to http://localhost:1234/debug/pprof/
```

Comment:
I'd mention following the setup/build guide within Reva since you need the binary for oCIS?

Reply:
Hm, how within Reva? You mean linking [here](https://github.com/cs3org/reva/#build-and-run-it-yourself)?

Reply:
Exactly. Probably it's obvious for most people but I'd be lost without at least some knowledge of how Golang/reva+oCIS work & play together